src/D2L.Bmx/Okta/OktaApi.cs

```diff
@@ -49,10 +49,15 @@ internal class OktaApi : IOktaApi {
 	}
 
 	async Task<AuthenticateResponse> IOktaApi.AuthenticateAsync( string username, string password ) {
-		var resp = await _httpClient.PostAsJsonAsync(
-			"authn",
-			new AuthenticateRequest( username, password ),
-			SourceGenerationContext.Default.AuthenticateRequest );
+		HttpResponseMessage resp;
+		try {
+			resp = await _httpClient.PostAsJsonAsync(
+				"authn",
+				new AuthenticateRequest( username, password ),
+				SourceGenerationContext.Default.AuthenticateRequest );
+		} catch( Exception ex ) {
+			throw new BmxException( "Okta authentication request failed. Try checking for a stable connection.", ex );
```

Comment:
should probably reword this to be broader. bad network connection isn't the only reason this can fail. It can also be Okta's fault (e.g. site down. it happened a few times)